pipelines/folioRancher/folioDevTools/moduleDeployment/deployModuleFromFeatureBranchEureka/Jenkinsfile

```diff
@@ -0,0 +1,268 @@
+
+/** Pipeline */
+ansiColor('xterm') {
+  // Common Gradle Options
+  String commonGradleOpts = '--quiet --console plain --no-daemon'
+
+  node(params.AGENT) {
+    try {
+      stage('Ini') {
+        buildName "#${params.MODULE_NAME}.${env.BUILD_ID}"
+        buildDescription "Env: ${namespace.getWorkspaceName()}\nBranch: ${params.MODULE_BRANCH}\n" + "Config: ${params.CONFIG_TYPE}\n${currentBuild.getBuildCauses()[0].shortDescription}"
+      }
+
+      stage('[Git] Checkout module source') {
+        checkout([
+          $class           : 'GitSCM',
+          branches         : [[name: "*/${featureBranch}"]],
+          extensions       : [[$class: 'RelativeTargetDirectory', relativeTargetDir: module.name],
+                              [$class: 'CleanBeforeCheckout', deleteUntrackedNestedRepositories: true],
+                              [$class: 'AuthorInChangelog'],
+                              [$class: 'SubmoduleOption', recursiveSubmodules: true]],
+          userRemoteConfigs: [[url: "https://github.com/folio-org/${module.name}.git"]]
+        ])
+      }
+
+      stage('[Maven/Gradle] Compile') {
+        // Check if Maven or Gradle build tool is used
+        module.buildTool = fileExists("${module.name}/pom.xml") ? 'maven' : 'gradle'
+
+        if (module.buildTool == 'maven') {
+          // Use specific Maven build directory
+          module.buildDir = "${env.WORKSPACE}/${module.name}"
+
+          dir(module.buildDir) {
+            // Read module version from pom.xml
+            module.version = readMavenPom(file: 'pom.xml').version
+
+            // Load Module Details for Maven builds
+            module.loadModuleDetails("${module.name}-${module.version}.${featureHash}")
+
+            // Build Module as Maven Project
+            withMaven(
+              jdk: "${common.selectJavaBasedOnAgent(params.AGENT)}".toString(),
+              maven: Constants.MAVEN_TOOL_NAME,
+              traceability: false,
+              options: [artifactsPublisher(disabled: true)])
+            {
+              sh """
+                mvn versions:set -DnewVersion=${module.getVersion()}
+                mvn package ${mavenArguments}
+              """.stripIndent()
+            }
+          }
+        }
+        else {
+          // Use specific Gradle build directory
+          module.buildDir = "${env.WORKSPACE}/${module.name}/service"
+
+          dir(module.buildDir) {
+            // Read module version from gradle.properties
+            module.version = readProperties(file: "gradle.properties")['appVersion']
+
+            // Load Module Details for Gradle builds
+            module.loadModuleDetails("${module.name}-${module.version}.${featureHash}")
+
+            // Build Module as Gradle Project
+            sh "./gradlew ${commonGradleOpts} -PappVersion=${module.version} assemble"
+          }
+        }
+      }
+
+      stage('[Docker] Build and push') {
```

Comment:
Same here, it's lot of code duplicates please refactor/optimise

Reply:
Code adjusted accordingly to your suggestion.
Please take a look on changes.